src/modules/ui.js

```diff
@@ -457,6 +462,37 @@ UIMenu.prototype.getEmote = function (name) {
 	return null;
 };
 
+UIMenu.prototype.doEmoteSearch = function (event) {
+	var self = this;
+
+	// Skip if in Edit mode
+	if (this.isEditing()) {
+		return;
+	}
+
+	// Get the searched value
+	var val = event.target.value.trim().toLowerCase();
+	
+	// Update visibility on emotes based on search
+	this.emotes.forEach(function (emote) {
+		// Skip if emote is marked hidden via Edit mode
+		if (!emote.isVisible()) {
+			return;
+		}
+
+		emote = self.getEmote(emote.getText());
+		var shouldShowEmote = emote.instance.getText().toLowerCase().includes(val);
+		emote.dom.toggleClass('emote-menu-hidden', !shouldShowEmote);
```

Comment:
when the emote menu is closed and reopened, what is the state of the menu? does search reset? do emotes reset?

Reply:
Indeed the states weren't consistent, pushed a change that clears input on menu close.